core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -26,74 +34,97 @@
         profiler.start("Docker compose container startup");
 
+        applyScaling(); // scale before up, so that all scaled instances are available first for linking
+        createServices();
+        registerContainersForShutdown();
+        startAmbassadorContainers(profiler);
+
+    }
+
+
+    private void createServices() {
         // Start the docker-compose container, which starts up the services
-        super.start();
-        followOutput(new Slf4jLogConsumer(logger()), OutputFrame.OutputType.STDERR);
+        getDockerCompose("up -d")
+                .start();
+    }
 
-        // wait for the compose container to stop, which should only happen after it has spawned all the service containers
-        logger().info("Docker compose container is running - service creation will start now");
-        while (isRunning()) {
-            logger().trace("Compose container is still running");
-            Uninterruptibles.sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
+    private DockerCompose getDockerCompose(String cmd) {
+        return new DockerCompose(composeFile, identifier)
+                .withCommand(cmd)
+                .withEnv(env);
+    }
+
+    private void applyScaling() {
+        // Apply scaling
+        if (!scalingPreferences.isEmpty()) {
+            StringBuilder sb = new StringBuilder("scale");
+            for (Map.Entry<String, Integer> scale : scalingPreferences.entrySet()) {
+                sb.append(" ").append(scale.getKey()).append("=").append(scale.getValue());
+            }
+
+            getDockerCompose(sb.toString())
+                    .start();
         }
-        logger().info("Docker compose has finished running");
+    }
 
+    private void registerContainersForShutdown() {
         // Ensure that all service containers that were launched by compose will be killed at shutdown
-        Filters namesContainingComposeIdentifier = new Filters().withFilter("name", "/" + identifier);
         try {
             List<Container> containers = dockerClient.listContainersCmd()
-                    .withFilters(namesContainingComposeIdentifier)
+                    .withLabelFilter("name=/" + identifier)
                     .withShowAll(true)
                     .exec();
 
-            // register with ContainerReaper to ensure final shutdown with JVM
-            containers.stream()
-                    .forEach(container ->
-                            ContainerReaper.instance().registerContainerForCleanup(container.getId(), container.getNames()[0]));
+            // register with ResourceReaper to ensure final shutdown with JVM
+            containers.forEach(container ->
+                    ResourceReaper.instance().registerContainerForCleanup(container.getId(), container.getNames()[0]));
+
+            // Ensure that the default network for this compose environment, if any, is also cleaned up
+            ResourceReaper.instance().registerNetworkForCleanup(identifier + "_default");
```

Comment:
ContainerReaper (now ResourceReaper) now cleans up compose-created networks too. This is done for the same reason as before - in case an uncontrolled shutdown prevents the `docker-compose down` being executed.